docs/docs/features/plugins.md

```diff
@@ -0,0 +1,276 @@
+     - in `pull` mode: Prometheus exposer port that should be accessable by Prometheus to pull the statistic;
+   - `mode`: interaction with Prometheus mode: `E_PUSH` or `E_PULL`;
+
+### Build and Run plugin in demo environment
+To build Prometheus plugin, pass '--plugins prometheus' argument to the build script, e.g.
+```bash
+bin/build-ubuntu.sh --plugins prometheus
+```
+To run plugin in demo environment, perform the following steps:
+
+1. Set plugin configuration:
+   - For `push` mode:
+   ```json
+       {
+         "host": "localhost",
+         "port": 9091,
+         "mode": "E_PUSH"
+       }
+   ```
+   - For `pull` mode:
+   ```
+       {
+         "host": "localhost",
+         "port": 8080,
+         "mode": "E_PULL"
+       }
+   ```
+
+2. Run BlazingMQ broker, it will automatically load and configure the plugin;
+
+3. Run Prometheus and Grafana services in Docker:
+   ```bash
+      docker compose -f docker/plugins/prometheus/docker-compose.yml up
+   ```
+
+4. In browser open link `http://localhost:9090/` with Prometheus UI to analyze available metrics;
+
+5. [Optional] In browser open link `http://localhost:3000/` with Grafana UI to analyze available metrics:
+   - Select `Prometrheus` data source;
+   - Set `http://prometheus:9090` as Prometheus server URL;
+
+### Integration tests
+Test plan:
+1. Test Prometheus plugin in 'push' mode:
+   - Run Prometheus (in docker);
+   - Run broker with local cluster and enabled Prometheus plugin in sandbox (temp folder);
+   - Put several messages into different queues;
+   - Request metrics from Prometheus and compare them with expected metric values.
+2. Test Prometheus plugin in 'pull' mode:
+   - Run Prometheus (in docker);
+   - Run broker with local cluster and enabled Prometheus plugin in sandbox (temp folder);
+   - Put several messages into different queues;
+   - Request metrics from Prometheus and compare them with expected metric values.
+
+Prerequisites:
+1. bmqbroker, bmqtool and prometheus plugin library should be built;
+2. Python3 should be installed;
+3. Docker should be installed, user launching the test script must be included into the group 'docker'.
+```bash
+Usage: ./src/plugins/prometheus/tests/prometheus_prometheusstatconsumer_test.py [-h] -p PATH
+options:
+  -h, --help            show this help message and exit
+  -p PATH, --path PATH  path to BlazingMQ build folder, e.g. './build/blazingmq'
+```
+
+### Labels
+The following labels may be used by each metric, whenever applicable. Refer to each individual metric documentation for restrictions which may apply.
+
+|Label Name|Description|
+|----------|-----------|
+|Instance|instance id of the broker (e.g. A), or 'default'|
+|Cluster|name of the cluster (e.g. bmqc01/A)|
+|Domain|name of the domain (e.g. my.domain)|
+|Tier|value of the tier (e.g. pd)|
+|Queue|name of the queue (e.g. my_queue)|
+|Role|role of the broker with regard to this queue (possible values are 'PRIMARY', 'PROXY', 'REPLICA', 'MEMBER')|
```

Comment:
A 'MEMBER' must be a 'PRIMARY' or a 'REPLICA'. Do we even need the 'MEMBER' role?

Reply:
No, we don't need it. We have only 3 roles now: 'PRIMARY', 'PROXY' and 'REPLICA'. It can also be 'UNKNOWN' if the queue hasn't been configured yet. However 'UNKNOWN' can't be posted as a tag. I think we should also update the GUTS docomentation on Tutty since this information has been taken from there

Reply:
Could you please also make a PR to update Tutti?

Reply:
Sure, please update in that PR